### PR TITLE
Improve documentation week12

### PR DESCRIPTION
**Pull request by pyhys**

### CODE REVIEW

File: src/device/pf_fspm.c [hunk 1]
```diff
@@ -1255,6 +1336,12 @@ int pf_fspm_signal_led_ind (pnet_t * net, bool led_state)
    int ret = 0;
    if (net->fspm_cfg.signal_led_cb != NULL)
    {
+      LOG_DEBUG (
```
Comment (olbjo) [inline src/device/pf_fspm.c:1339]: What is the intended use of the log levels? These are all log entries resulting from api calllabacks. Maybe they should be of higher log level info? I do not know, the application will problably create its own log entries for these callbacks anyway.

**Review by olbjo — APPROVED** [review]
Great work!
Some minor comments, reject them if you disagree.

File: doc/implementation_details.rst [hunk 2]
```diff
@@ -718,6 +718,8 @@ Show lots of details of the stack state::
 
 Coding rules
 ------------
+Readablility counts.
```
[the remark this answers is not in the document]
Comment (pyhys) [reply doc/implementation_details.rst]: Removed

File: sample_app/sampleapp_common.c [hunk 3]
```diff
@@ -422,7 +422,7 @@ static int app_connect_ind (
 
    if (p_appdata->arguments.verbosity > 0)
    {
-      printf ("Connect call-back. AREP: %u\n", arep);
+      printf ("PLC is connecting. AREP: %u\n", arep);
```
[the remark this answers is not in the document]
Comment (pyhys) [reply sample_app/sampleapp_common.c:425]: The sample app is a pedagogical tool, and the fact that it is an external stimuli is important.

File: sample_app/sampleapp_common.c [hunk 4]
```diff
@@ -463,7 +463,7 @@ static int app_dcontrol_ind (
    if (p_appdata->arguments.verbosity > 0)
    {
       printf (
-         "Dcontrol call-back. AREP: %u  Command: %s\n",
+         "PLC has sent DControl message. AREP: %u  Command: %s\n",
```
[the remark this answers is not in the document]
Comment (pyhys) [reply sample_app/sampleapp_common.c:466]: See above

File: sample_app/sampleapp_common.c [hunk 5]
```diff
@@ -482,7 +482,7 @@ static int app_ccontrol_cnf (
    if (p_appdata->arguments.verbosity > 0)
    {
       printf (
-         "Ccontrol confirmation call-back. AREP: %u  Status codes: %d %d %d "
+         "PLC has confirmed our CControl message. AREP: %u  Status codes: %d %d %d "
```
[the remark this answers is not in the document]
Comment (pyhys) [reply sample_app/sampleapp_common.c]: See above

File: sample_app/sampleapp_common.c [hunk 6]
```diff
@@ -512,7 +512,7 @@ static int app_write_ind (
    if (p_appdata->arguments.verbosity > 0)
    {
       printf (
-         "Parameter write call-back. AREP: %u API: %u Slot: %2u Subslot: %u "
+         "PLC will write record. AREP: %u API: %u Slot: %2u Subslot: %u "
```
[the remark this answers is not in the document]
Comment (pyhys) [reply sample_app/sampleapp_common.c:515]: See above

File: sample_app/sampleapp_common.c [hunk 7]
```diff
@@ -594,7 +594,7 @@ static int app_read_ind (
    if (p_appdata->arguments.verbosity > 0)
    {
       printf (
-         "Parameter read call-back. AREP: %u API: %u Slot: %2u Subslot: %u "
+         "PLC will read record. AREP: %u API: %u Slot: %2u Subslot: %u "
```
[the remark this answers is not in the document]
Comment (pyhys) [reply sample_app/sampleapp_common.c:597]: See above

File: src/device/pf_cmio.c [hunk 8]
```diff
@@ -164,7 +165,7 @@ int pf_cmio_cmdev_state_ind (
 
    LOG_DEBUG (
       PNET_LOG,
-      "CMIO(%d): Received event %s from CMDEV. Our state %s. AREP %u\n",
+      "CMIO(%d): Received event %s from CMDEV. Our initial state %s for AREP %u.\n",
```
[the remark this answers is not in the document]
Comment (pyhys) [reply src/device/pf_cmio.c]: Done

File: src/device/pf_cmpbe.c [hunk 9]
```diff
@@ -83,22 +83,28 @@ void pf_cmpbe_show (const pf_ar_t * p_ar)
...
+   }
    p_ar->cmpbe_state = state;
 }
 
 int pf_cmpbe_cmdev_state_ind (pf_ar_t * p_ar, pnet_event_values_t event)
 {
    LOG_DEBUG (
       PNET_LOG,
-      "CMPBE(%d): Received event %s from CMDEV. Our state %s.\n",
+      "CMPBE(%d): Received event %s from CMDEV. Our initial state %s for AREP %u.\n",
```
[the remark this answers is not in the document]
Comment (pyhys) [reply src/device/pf_cmpbe.c]: Done

File: src/device/pf_cmsm.c [hunk 10]
```diff
@@ -167,7 +168,7 @@ int pf_cmsm_cmdev_state_ind (
 
    LOG_DEBUG (
       PNET_LOG,
-      "CMSM(%d): Received event %s from CMDEV. Our state %s. AREP %u\n",
+      "CMSM(%d): Received event %s from CMDEV. Our initial state %s for AREP %u.\n",
```
[the remark this answers is not in the document]
Comment (pyhys) [reply src/device/pf_cmsm.c]: Done

File: src/device/pf_cmsu.c [hunk 11]
```diff
@@ -61,6 +100,14 @@ int pf_cmsu_cmdev_state_ind (
 {
    uint32_t crep;
 
+   LOG_DEBUG (
+      PNET_LOG,
+      "CMSU(%d): Received event %s from CMDEV. Our initial state %s for AREP %u.\n",
```
[the remark this answers is not in the document]
Comment (pyhys) [reply src/device/pf_cmsu.c]: Done

File: src/device/pf_fspm.c (hunk 1 above)
Comment (pyhys) [reply src/device/pf_fspm.c:1409]: I have found that when dissecting log files from users, it is very handy to know when the callbacks have been triggered. If callbacks are missing is also important information. This is mainly a debug feature, so I think the level is appropriate.

File: doc/getting_started_rtkernel.rst [hunk 12]
```diff
@@ -289,6 +289,50 @@ You might need to press the Play button in the Workbench if you have enabled
...
+
+In the root folder of the rt-kernel directory::
+
+    source setup.sh integrator
+    make clean
+    make -j
+
+In the parent directory of ``p-net``, configure a new build directory::
+
+    RTK=<PATH TO YOUR MODIFIED>/rt-kernel-xmc4 \
```
Comment (hefloryd) [inline doc/getting_started_rtkernel.rst:308]: xmc4 should probably be removed here.

**Review by hefloryd — APPROVED** [review]

File: doc/getting_started_rtkernel.rst (hunk 12 above)
Comment (pyhys) [reply doc/getting_started_rtkernel.rst]: Done

**Pull request merged**